libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -180,6 +183,43 @@ bool SyntaxChecker::visit(Throw const& _throwStatement)
 	return true;
 }
 
+bool SyntaxChecker::visit(Literal const& _literal)
+{
+	if (!_literal.isHexNumber())
+		return true;
+	// We have a hex literal. Do underscore validation
```

Comment:
Why isn't this code also validating for decimals?

Reply:
@axic For integers, fractional and scientific number with decimal digits, scanner itself takes care of underscores. For hex, SyntaxChecker validates. We could move both to SyntaxChecker I guess, but underscore validation for numbers with . and scientific notation would be complicated.